### PR TITLE
Add deferred named version

### PR DESCRIPTION
**Pull request by sanohin**
**What**: added named deferred version as probably many users need such macro. Added tests and readme for this case.

**How**: seems like handled all the cases.

**Checklist**:

* [x] Documentation
* [x] Tests
* [ ] Ready to be merged 
* [ ] Added myself to contributors table

### CODE REVIEW

**Review by kentcdodds — COMMENTED** [review]
Thanks, but I'm going to need a little convincing that this is a good idea. My first impression is that this is stepping a little beyond the bounds of how I want import-all.macro to work...

**Comment by sanohin** [comment]
Well, It would be difficult to convince you probably. While working in few different companies I saw the approach of exposing 'kinda' library additional components or functions to globals and using them somewhere dynamically (component/fn name cannot be statically determined). 

So I think this can be a great drop-in replacement for such use cases with no changes to the architecture of the whole app.

File: src/macro.js [hunk 1]
```diff
@@ -129,6 +147,99 @@ function deferredVersion({referencePath, state, babel}) {
...
+      opts: {filename},
+    },
+  } = state
+  const fileDir = path.dirname(filename)
+  const importSources = getImportSources(
+    referencePath.parentPath.parentPath,
+    fileDir,
+  )
+
+  const getAst = code => {
```
Comment (sanohin) [inline src/macro.js:163]: btw, not sure how to get covered this function by tests

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Sorry, I think I'm going to close this one. You can make your own macro for your projects.

Thanks anyway!

**Pull request closed without merging**